sympy/combinatorics/rewritingsystem.py

```diff
@@ -285,3 +298,190 @@ def reduce(self, word, exclude=None):
+                current_state_name_array = []
+                next_sub = current_state_name
+                while len(next_sub):
+                    current_state_name_array.append(next_sub)
+                    next_sub = next_sub.subword(1, len(next_sub))
+                # Check if the transition to any new state in posible.
+                for sub_word in current_state_name_array:
+                    if sub_word in accept_states:
+                        # Add accept states.
+                        if current_state_type == "accept":
+                            for letter in automaton_alphabet:
+                                next = current_state_name*letter
+                                while True:
+                                    if len(next) <= 1:
+                                        if next in self.reduction_automaton.states:
+                                            self.reduction_automaton.states[state].add_transition(letter, next)
+                                        else:
+                                            self.reduction_automaton.states[state].add_transition(letter, 'start')
+                                        break
+                                    else:
+                                        if next in self.reduction_automaton.states:
+                                            self.reduction_automaton.states[state].add_transition(letter, next)
+                                            break
+                                        next = next.subword(1, len(next))
+                        break
+
+        # Add transitions for new states. All symbols used in the automaton are considered here.
+        # Ignore this if `generators` = `automaton_alphabet`.
+        if len(generators) != len(automaton_alphabet):
+            for state in proper_prefixes:
+                for letter in generators:
+                    next = current_state_name*letter
+                    len_next_word = len(next)
+                    while True:
+                        if len(next) <= 1:
+                            if next in self.reduction_automaton.states:
+                                self.reduction_automaton.states[state].add_transition(letter, next)
+                            else:
+                                self.reduction_automaton.states[state].add_transition(letter, 'start')
+                            break
+                        else:
+                            if next in self.reduction_automaton.states:
+                                self.reduction_automaton.states[state].add_transition(letter, next)
+                                break
+                            next = next.subword(1, len(next))
+
+    def reduce_using_automaton(self, word):
+        '''
+        Reduce a word using an automaton.
+
+        Summary:
+        All the elements of the automaton are stored in an array and are given as the input to the automaton.
```

Comment:
I'm not sure what this sentence means. Could you try to say it a bit more clearer? For example, the input to the automaton is presumably the word we are reducing, not an array of elements of the automaton. Also, what do we mean by "elements"? States or letters of the alphabet or something else?

Reply:
That needs to be changed. I think I forgot to update the docstring for this method. Well, making it "All the symbols of the given word are stored in an array..."